gui/blueprint.lua

```diff
@@ -82,6 +82,19 @@ function BlueprintUI:onAboutToShow()
     if not dfhack.isMapLoaded() then
         qerror('Please load a fortress map.')
     end
+
+    self.saved_mode = df.global.ui.main.mode
+    if dfhack.gui.getCurFocus(true):find('^dfhack/') then
+        self.saved_mode = df.ui_sidebar_mode.Default
+    end
+    guidm.enterSidebarMode(df.ui_sidebar_mode.LookAround)
+end
+
+function BlueprintUI:onDismiss()
+    if not guidm.SIDEBAR_MODE_KEYS[self.saved_mode] then
+        self.saved_mode = df.ui_sidebar_mode.Default
+    end
+    guidm.enterSidebarMode(self.saved_mode)
```

Comment:
I can't seem to get this to work - pressing `v`, then running `gui/blueprint`, then pressing `Esc` at any stage always seems to bring me back to the default sidebar mode, rather than the `v` mode. I don't think this is super important, but wanted to confirm that I wasn't missing something obvious.

Reply:
Yeah, not all modes are handled. We can extend the map in enterSidebarMode over time.

Reply:
Oh, I thought that was one that was handled. Looks like it's not. Thanks!

Reply:
it totally should have been handled. Added in DFHack/dfhack#1889